DSCResources/MSFT_xIisFeatureDelegation/MSFT_xIisFeatureDelegation.psm1

```diff
@@ -0,0 +1,160 @@
+######################################################################################
+# DSC Resource for IIS Server level Feature Delegation
+######################################################################################
+
+data LocalizedData
+{
+    # culture="en-US"
+    ConvertFrom-StringData @'
+NoWebAdministrationModule=Please ensure that WebAdministration module is installed.
+UnableToGetConfig=Unable to get configuration data for '{0}'
+ConfigNotFound=No configuration data found for '{0}'
+ChangedMessage=Changed overrideMode for '{0}' to {1}
+'@
+}
+
+######################################################################################
+# The Get-TargetResource cmdlet.
+# This function will get the Mime type for a file extension
+######################################################################################
+function Get-TargetResource
+{
+  [OutputType([Hashtable])]
+	param
+	(		
+		[Parameter(Mandatory)]
+		[ValidateNotNullOrEmpty()]
+		[String]$SectionName,
+
+		[Parameter(Mandatory)]
+		[ValidateNotNullOrEmpty()]
+        [ValidateSet("Allow", "Deny")]
+        [String]$OverrideMode
+	)
+	
+    CheckIISPoshModule
+
+    $getTargetResourceResult = $null;
+
+    [string]$oMode = GetOverrideMode $SectionName
+
+    if ($oMode -eq $null)
+    {
+        $ensureResult = "Absent";
+    }
+    else
+    {
+        $ensureResult = "Present"
+        $getTargetResourceResult = @{SectionName = $SectionName
+                                    OverrideMode = $oMode}
+    }
+
+    
+	return $getTargetResourceResult
```

Comment:
Given you have mandatory parameters that can't be null or empty it seems strange that you would be returning a a possible null here. I would move the $getTargetResourceCall outside of the if/else, remove the assignment to null and just have it return at the end of the end of the function so it always returns the hashtable with some data.